system-int-test/test/cmr/system_int_test/search/collection_geojson_search_test.clj

```diff
@@ -0,0 +1,134 @@
+(ns cmr.system-int-test.search.collection-geojson-search-test
+  (:require
+    [clojure.test :refer :all]
+    [clojure.java.io :as io]
+    [cmr.common.util :as util :refer [are3]]
+    [cmr.common-app.test.side-api :as side]
+    [cmr.search.services.parameters.converters.shapefile :as shapefile]
+    [cmr.spatial.line-string :as l]
+    [cmr.spatial.mbr :as m]
+    [cmr.spatial.point :as p]
+    [cmr.spatial.polygon :as poly]
+    [cmr.system-int-test.data2.core :as d]
+    [cmr.system-int-test.data2.granule :as dg]
+    [cmr.system-int-test.data2.umm-spec-collection :as data-umm-c]
+    [cmr.system-int-test.utils.index-util :as index]
+    [cmr.system-int-test.utils.ingest-util :as ingest]
+    [cmr.system-int-test.utils.search-util :as search]
+    [cmr.umm.umm-spatial :as umm-s]
+    [cmr.system-int-test.data2.collection :as dc]))
+
+(use-fixtures :each (ingest/reset-fixture {"provguid1" "PROV1"}))
+
+(defn- polygon
+  "Creates a single ring polygon with the given ordinates. Points must be in counter clockwise
+  order."
+  [& ords]
+  (poly/polygon [(apply umm-s/ords->ring ords)]))
+
+(defn- make-coll
+  [coord-sys et & shapes]
+  (let [shapes (map (partial umm-s/set-coordinate-system coord-sys) shapes)]
+    (d/ingest "PROV1"
+              (dc/collection
+               {:entry-title et
+                :spatial-coverage (dc/spatial {:gsr coord-sys
+                                               :sr coord-sys
+                                               :geometries shapes})}))))
+
+(deftest collection-geojson-search-test
```

Comment:
Feel free to ignore this comment - only do this if it makes things easier for when you add KML.

It seems like you might have the same test setup for all the different shapefile tests (and maybe some of the other spatial tests). It might be possible to write most of the setup and testing in one place and then just pass in the mime-type and extension.

So this test would become:
```
(deftest collection-geojson-search-test
  (side/eval-form `(shapefile/set-enable-shapefile-parameter-flag! true)
  (shapefile-spec-test 'geojson' 'application/geo+json')
```

Reply:
I'll do this as part of the KML ticket.

Reply:
I'll do this as part of the KML ticket.